through2-batch.js

```diff
@@ -42,11 +42,16 @@ module.exports = function batchThrough(options, transform, flush) {
       var self = this;
 
       if (batched.length > 0) {
-          transform.call(this, batched, lastEnc, function (err) {
+          if (transform) {
+              transform.call(this, batched, lastEnc, function (err) {
+                  batched = [];
+                  if (err) callback(err);
+                  else flush.call(self, callback);
+              });
+          } else {
               batched = [];
```

Comment:
You should probably call `callback(null, batched);` before throwing the batch away.

This way, the last batch is just dropped without notice.

```suggestion
              callback(null, batched);
              batched = [];
```